test/lib/foreman/access_permissions_test.rb

```diff
@@ -44,6 +44,8 @@ class AccessPermissionsTest < ActiveSupport::TestCase
 
   MAY_SKIP_AUTHORIZED = [ "about/index" ]
 
+  SPECIAL_PATH = ['api/v2/puppet_hosts/puppetrun']
```

Comment:
It's a global constant that holds all actions that have special paths - these actions cannot be tested by this test. [This line](https://github.com/ShimShtein/foreman/blob/73fe3dc47c3faee6d44ced1f059bf9cd5dce176d/test/lib/foreman/access_permissions_test.rb#L51) generates paths from controller names, and in this case it's not going to work.